src/mainwindow.h

```diff
@@ -31,15 +31,17 @@ class MainWindow : public QMainWindow
 
 private slots:
     void toggleFullScreen();
-    void when_ReadingList_toggled(bool state);
-    void when_libraryPageDisplayed(bool showed);
+    void tabChanged(bool libraryDisplayed);
+    void readingListToggled(bool state);
+    void libraryPageDisplayed();
     void hideTabAndTop();
     void showTabAndTop();
 
 private:
     Ui::MainWindow *mp_ui;
     About     *mp_about;
     LocalKiwixServer *mp_localKiwixServer;
+    bool readingListRemainActive = false;
```

Comment:
Do we really need this hidden state variable? Can't the state of the reading list be fully maintained in the `ToggleReadingListAction` action (so that it stays checked even when switching to tabs alongside which the reading list isn't supposed to be displayed).

Reply:
I understand that but, IMHO, the checked state of the reading list button can serve the same purpose and this new variable violates the DRY principle. Whether the reading list should be displayed on the library and/or settings tab is a different matter. If the users understand that the reading list is visible only on content tabs, they can easily accept the fact that the reading list button is pressed but there is no reading list.

Reply:
I agree and I can see where the confusion is coming from because I was thinking the same when I started developing this. The existence of this variable derives from the fact that the reading list shouldn't be displayed on the library. Let me explain.

Specifically, when `readingListRemainActive` is `true` it denotes that the library is hidden but should remain active on the first possible moment. This is why the value of it when navigating *inside* the library becomes true.

 The reason we cannot be reliant on the checked state of the reading list button is becausem when entering the library, that is checked to false. So, if we don't have `readingListRemainActive` and a user activates the reading list and enters the library the reading list checked value is going to be set to false. After navigating away from the library we have no way to know the if the reading list was active to keep it open.

I hope this clears things up a bit and is not confusing.

Reply:
> The reason we cannot be reliant on the checked state of the reading list button is because when entering the library, that is checked to false.

That's exactly what I am proposing to change. The reading list button must be checked/unchecked solely by the user. In other words when the reading list button is checked and the current tab changes to library or settings the reading list button stays checked (but the sidebar is update accordingly). I think that such behavior will be less confusing once the users realize that the reading list is displayed only alongside the content tabs.

Reply:
In that case, I totally agree. It seems to me also that it makes sense the most.